force_wfmanager/left_side_pane/workflow_tree.py

```diff
@@ -235,131 +111,342 @@ class WorkflowTree(ModelView):
+                                      None, 'DataSources')
                 ),
-            VGroup(
-                UReadonly(
-                    name='selected_error',
-                    editor=TextEditor(),
+            ],
+            orientation="horizontal",
+            editable=False,
+            selected="selected_mv",
+        )
+
+        view = View(
+            Group(
+                VGroup(
+                    UItem(name='workflow_mv',
+                          editor=tree_editor,
+                          show_label=False
+                          ),
+                ),
+                VGroup(
+                    UReadonly(
+                        name='selected_error',
+                        editor=TextEditor(),
                     ),
-                label='Workflow Errors',
-                show_border=True
+                    label='Workflow Errors',
+                    show_border=True
                 ),
             ),
-        width=800,
-        height=600,
-        resizable=True,
+            width=500,
+            resizable=True,
         )
 
-    def __init__(self, model, factory_registry, *args, **kwargs):
-        super(WorkflowTree, self).__init__(*args, **kwargs)
-        self.model = model
-        self._factory_registry = factory_registry
+        return view
 
     def _workflow_mv_default(self):
         return WorkflowModelView(model=self.model)
 
+    # TODO: Add in workflow status screen
+    def show_workflow_status(self, name):
+        pass
+
     @on_trait_change('model')
     def update_model_view(self):
         """Update the workflow modelview's model and verify, on either loading
-        a new workflow, or an internal change to the workflow"""
-
+        a new workflow, or an internal change to the workflow.
+        """
         self.workflow_mv = WorkflowModelView(model=self.model)
         self.verify_workflow_event = True
 
-    @triggers_verify
-    def new_mco(self, ui_info, object):
-        """ Opens a dialog for creating a MCO """
-
-        workflow_mv = self.workflow_mv
-
-        modal = NewEntityModal(factories=self._factory_registry.mco_factories)
-        modal.edit_traits()
-        result = modal.current_model
-
-        if result is not None:
-            workflow_mv.set_mco(result)
-
-    @triggers_verify
-    def edit_mco(self, ui_info, object):
-        ModelEditDialog(model=object.model).edit_traits()
+    def selection(self, factory_name, instance_name, object):
+        """Assigns the correct add/remove entity functions based on which
+        object is selected in the workflow tree"""
+        instance_select_function = {
+            'Notification Listeners': self.delete_notification_listener,
+            'MCO': self.delete_mco, 'KPIs': self.delete_kpi,
+            'Execution Layers': self.delete_layer,
+            'DataSources': self.delete_data_source,
+            'Parameters': self.delete_parameter
+        }
+        factory_select_function = {
```

Comment:
It's kind of annoying having to have this, but the requirements on selecting different factories are just about different enough that having a generic function would be harder to read.